Update sketches of WiFi-ISM43362-M3G-L44 Arduino library

biagiom opened the pull request:
- Fix some whitespace errors
- Improve documentation and comments
- Fix printing of MAC address and BSSID in some examples
- Improve code of some sketches

biagiom committed:
Update sketches of WiFi-ISM43362-M3G-L44 Arduino library

- Fix some whitespace errors
- Improve documentation and comments
- Fix printing of MAC address and BSSID in some examples
- Improve code of some sketches

fpistm commented:
Hi @biagiom  and thanks for those fixes.
I've saw the print MAC issue and planned to fix it. ;)

fpistm commented on examples/ConnectWithWEP/ConnectWithWEP.ino:
```diff
@@ -1,19 +1,20 @@
-
+ * ConnectWithWEP 
+ *
+ * This example is used to connects to a WEP-encrypted WiFi network.
```
connect~s~

fpistm commented on examples/ConnectWithWEP/ConnectWithWEP.ino:
```diff
@@ -45,22 +46,22 @@ int keyIndex = 0;                                // your network key Index numbe
+    Serial.println("WiFi module not detected");
     // don't continue:
-    while (true);
+    while (true) ;
```
Why adding a space here?

fpistm commented on examples/ConnectNoEncryption/ConnectNoEncryption.ino:
```diff
@@ -91,61 +92,91 @@ void printWifiData() {
+  printEncryptionType(WiFi.encryptionType());
+  Serial.println();
+}
+
```
line could be removed

fpistm commented on examples/ConnectWithWEP/ConnectWithWEP.ino:
```diff
@@ -96,52 +97,80 @@ void printWifiData() {
+      Serial.println("WPA_TKIP");
+      break;
+     case ES_WIFI_SEC_UNKNOWN:
+      Serial.println("UNKNOW");
```
UNKNOWN

fpistm commented on examples/ConnectNoEncryption/ConnectNoEncryption.ino:
```diff
@@ -91,61 +92,91 @@ void printWifiData() {
+      Serial.println("WPA_TKIP");
+      break;
+     case ES_WIFI_SEC_UNKNOWN:
+      Serial.println("UNKNOW");
```
UNKNOWN

fpistm commented on examples/ConnectWithWPA/ConnectWithWPA.ino:
```diff
@@ -1,11 +1,12 @@
+ *
  */
-
+ 
```
remove extra space

fpistm commented on examples/ConnectWithWPA/ConnectWithWPA.ino:
```diff
@@ -29,27 +30,27 @@ Configure WiFi:
+    Serial.println("WiFi module not detected");
     // don't continue:
-    while (true);
+    while (true) ;
```
Why an extra space?

fpistm commented on examples/ConnectWithWPA/ConnectWithWPA.ino:
```diff
@@ -87,53 +88,80 @@ void printWifiData() {
+      Serial.println("WPA_TKIP");
+      break;
+     case ES_WIFI_SEC_UNKNOWN:
+      Serial.println("UNKNOW");
```
UNKNOWN

fpistm commented on examples/ConnectWithWPA/ConnectWithWPA.ino:
```diff
@@ -87,53 +88,80 @@ void printWifiData() {
+     case ES_WIFI_SEC_WPA2_TKIP:
+      Serial.println("WPA_TKIP");
+      break;
+     case ES_WIFI_SEC_UNKNOWN:
```
Add `default:`

fpistm commented on examples/ConnectWithWEP/ConnectWithWEP.ino:
```diff
@@ -96,52 +97,80 @@ void printWifiData() {
+     case ES_WIFI_SEC_WPA2_TKIP:
+      Serial.println("WPA_TKIP");
+      break;
+     case ES_WIFI_SEC_UNKNOWN:
```
Add `default:`

fpistm commented on examples/ConnectNoEncryption/ConnectNoEncryption.ino:
```diff
@@ -91,61 +92,91 @@ void printWifiData() {
+     case ES_WIFI_SEC_WPA2_TKIP:
+      Serial.println("WPA_TKIP");
+      break;
+     case ES_WIFI_SEC_UNKNOWN:
```
Add `default:`

fpistm commented on examples/ScanNetworks/ScanNetworks.ino:
```diff
@@ -34,57 +34,56 @@ SPIClass SPI_3(PC12, PC11, PC10);
   }
 
-  // Print WiFi MAC address:
+  // Print WiFi MAC addresses:
```
only one address print.

fpistm commented on examples/ScanNetworks/ScanNetworks.ino:
```diff
@@ -93,29 +92,29 @@ void listNetworks() {
   if (numSsid == -1) {
     Serial.println("Couldn't get a wifi connection");
-    while (true);
+    while (true) ;
```
ditto

fpistm commented on examples/ScanNetworks/ScanNetworks.ino:
```diff
@@ -93,29 +92,29 @@ void listNetworks() {
+void printEncryptionType(uint8_t encryptionType) {
   // read the encryption type and print out the name:
-  switch (thisType) {
+  switch (encryptionType) {
```
same:
Add `default:`
and UNKNOWN

fpistm commented on examples/WiFiWebClientRepeating/WiFiWebClientRepeating.ino:
```diff
@@ -53,37 +53,30 @@ void setup() {
-  // if there's incoming data from the net connection.
-  // send it out the serial port.
+  // if there's incoming data from the server,
+  // print it out to the Serial Monito.
```
Monitor

fpistm reviewed (changes requested):
Thanks for all those corrections.
Still some to fix, thanks in advance.

biagiom committed:
Add improvements to some examples of WiFi-ISM43362-M3G-L44 library

- Fix some spelling errors
- Remove some extra spaces
- Add default: case in switch() statement for sketches
  that contain the printEncryptionType() function

biagiom commented:
Hi @fpistm and many thanks for your review.
First of all I world like to say a big thank you to all the ST team members involved in the development and maintenance of the STM32duino project (you, @LMESTM, @fprwi6labs, @cparata, @VVESTM, ... other?), in fact I think that it is a very useful and fast way to develop a project with STM32 boards.
Secondly, I have added the fixes you have pointed out but if there are any other fixes to add, let me know.
Best regards.
Biagio.

LMESTM commented:
@biagiom thanks a lot for your nice comment.
@fpistm is off just now, but he will follow-up when he's back next week - let me know if this ok to wait until he's back or if I need to take over

biagiom commented:
Hi @LMESTM, for me it's ok to wait until @fpistm will come back, however if you want to review the changes feel free to do that.
Best regards.

LMESTM commented on examples/ConnectNoEncryption/ConnectNoEncryption.ino:
```diff
@@ -91,61 +92,92 @@ void printWifiData() {
+    if (i != 5) {
+      Serial.print(":");
+    }
+    else {
```
use either
```
}
else
{
```
or 
```
} else {
```

biagiom replied:
it's ok if I use the following coding style for all the examples ?
```
if (condition) {
  // do something
}
```
```
if (condition) {
  // do something
} else {
  // do other things
}
```
```
if (condition) {
  // do something
} else if (other_condition) {
  // do other things
} else {
 // ...
}
```

fpistm replied:
yes, fine

LMESTM commented on examples/ConnectNoEncryption/ConnectNoEncryption.ino:
```diff
@@ -91,61 +92,92 @@ void printWifiData() {
-  Serial.print(mac[1], HEX);
-  Serial.print(":");
-  Serial.println(mac[0], HEX);
+  for (uint8_t i = 0; i < 6; i++) {
```
Before this patch MAC is printed from mac[5] to mac[0], 
now it's printed from 0 to 5 - is it intended ?

fpistm replied:
yes, it was inverted before. I've mentioned it in my first comment of this PR.

LMESTM commented on examples/ConnectNoEncryption/ConnectNoEncryption.ino:
```diff
@@ -91,61 +92,92 @@ void printWifiData() {
-  Serial.print(":");
-  Serial.println(bssid[0], HEX);
+  for (uint8_t i = 0; i < 6; i++) {
+    if (bssid[i] < 0x10) {
```
same as before about order of prints

LMESTM reviewed (commented):
I only reviewed first example - see only minor feedback. 
Thanks a lot for improving the quality of those examples!

fpistm commented on examples/ConnectNoEncryption/ConnectNoEncryption.ino:
```diff
@@ -91,61 +92,92 @@ void printWifiData() {
+    case ES_WIFI_SEC_WPA2_TKIP:
+      Serial.println("WPA_TKIP");
+      break;
+    case ES_WIFI_SEC_UNKNOWN:
```
You  could do it like this to avoid duplicate `Serial.println`
```
+    case ES_WIFI_SEC_UNKNOWN:
+    default:
+      Serial.println("UNKNOWN");
+      break;
```

biagiom replied:
Ok, I will add your suggestions in the next commit

fpistm commented:
You're welcome @biagiom 
Thanks also to contribute, every help is greatly appreciated ;)
Even if I'm OoO, I follow the stm32duino activities which is bigger and bigger

biagiom committed:
Improve examples of WiFi-ISM43362-M3G-L44 library

- Improve code formatting
- Improve printing of network information in some sketches

biagiom commented:
Hi @fpistm,
I added a new commit to include your suggestions.
Let me known if I forgot something or I have to add other changes.
Regards.
